app/queries/get_activity_log_events.rb

```diff
@@ -15,25 +15,44 @@ class GetActivityLogEvents
     ],
   }.freeze
 
-  CHANGES_TO = %w[
+  INCLUDE_APPLICATION_CHOICE_CHANGES_TO = %w[
     reject_by_default_feedback_sent_at
     offer_changed_at
   ].freeze
 
+  IGNORE_STATUS = %i[interviewing].freeze
+
   def self.call(application_choices:, since: nil)
     since ||= Time.zone.local(2018, 1, 1) # before the pilot began, i.e. all records
 
-    filter = CHANGES_TO.map { |f| "jsonb_exists(audited_changes, '#{f}')" }
-                      .join(' OR ')
+    application_choice_audits_filter = INCLUDE_APPLICATION_CHOICE_CHANGES_TO.map { |f|
+      "jsonb_exists(audited_changes, '#{f}')"
+    }.join(' OR ')
+
+    filtered_statuses = ApplicationStateChange::STATES_VISIBLE_TO_PROVIDER - IGNORE_STATUS
+    status_transitions_to_include = filtered_statuses.map { |status| "'#{status}'" }.join(', ')
+
+    application_choice_audits_filter += " OR audited_changes::json#>>'{status, 1}' IN (#{status_transitions_to_include})"
+
+    Audited::Audit.includes(INCLUDES).from <<~COMBINE_AUDITS_WITH_APPLICATION_CHOICES_SCOPE_AND_FILTER.squish
+      (
+        SELECT a.*
+          FROM audits a
 
-    statuses_visible_to_providers = ApplicationStateChange::STATES_VISIBLE_TO_PROVIDER.map { |status| "'#{status}'" }.join(', ')
-    filter += " OR audited_changes::json#>>'{status, 1}' IN (#{statuses_visible_to_providers})"
+          INNER JOIN (#{application_choices.to_sql}) ac
+            ON (
+              auditable_type = 'ApplicationChoice'
+              AND auditable_id = ac.id
+              AND action = 'update'
+              AND ( #{application_choice_audits_filter} )
+            ) OR (
+              associated_type = 'ApplicationChoice'
+              AND associated_id = ac.id
+            )
 
-    Audited::Audit.includes(INCLUDES)
-                  .where(auditable: application_choices)
-                  .where(action: :update)
-                  .where(filter)
-                  .where('created_at > ?', since)
-                  .order(created_at: :desc)
+          WHERE a.created_at >= '#{since.iso8601}'::TIMESTAMPTZ
+          ORDER BY a.created_at DESC
+      ) AS audits
+    COMBINE_AUDITS_WITH_APPLICATION_CHOICES_SCOPE_AND_FILTER
```

Comment:
I wonder if the join could lead to duplicates. Thinking about it probably not assuming there are no audit records with `associated` values the same as `auditable`.

Reply:
good question, might be worth dropping a distinct in there just incase

Reply:
This is a good thing to consider, but the answer is duplicates are not possible here, because:

1. Application choice ids are unique (unless there's a problem with the `GetApplicationChoicesForProviders` query)

2. All `audits` considered will have either `auditable_type = 'ApplicationChoice'` or `associated_type = 'ApplicationChoice` from normal use of the app.

3. Even if you artificially constructed an `Audit` record with both `auditable` and `associated` pointing to the same _application choice_ (which makes no sense), you wouldn't get duplicates because the join evaluates records against each other, so this `Audit` would only be considered once.

4. Yes, you could get duplicates by artificially crafting an `Audit` with `auditable` pointing to one application choice and `associated` pointing to another application choice, both of which are part of the `GetApplicationChoicesForProviders` result set.

Including a `DISTINCT` is, in theory, a good idea, but it can have a serious impact on performance, so I'll leave it out and make sure that `GetApplicationChoicesForProviders` returns no duplicates when we change it.

Reply:
By duplicates, I assume you mean the same audit event being shown more than once. See also: https://stackoverflow.com/questions/40218138/will-inner-join-allow-duplicates